File: programmers/[고득점Kit]브루트포스/소수찾기.py
from itertools import permutations

def sosu(num):
    t = 2
    if num < 2:
        return False

    while t * t <= num:
        if num % t == 0:
            return False
        t += 1
    return True


def solution(numbers):
    answer = []

    for i in range(1, len(numbers) + 1):
        check = list(map(''.join, permutations(numbers, i)))
        for j in list(set(check)):
            if sosu(int(j)):
                answer.append(int(j))

    answer = set(answer)
    return len(answer)

File: By-Swift/By-Swift/Programmers/구현/소수찾기.swift
import Foundation

func combinations(_ array: [String]) -> Set<String> {
    if array.count == 0 { return [] }

    let answerArray = (0..<array.count).flatMap { i -> [String] in
        var removedArray = array
        let elem = removedArray.remove(at: i)
        return [elem] + combinations(removedArray).map { elem + $0 }
    }

    return Set(answerArray)
}

func sosu(_ num: Int) -> Bool {
    var check = 2
    if num < 2 { return false }
    
    while (check*check) <= num {
        if num % check == 0 { return false }
        check += 1
    }
    return true
}

func solution(_ numbers:String) -> Int {
    let arrs = combinations(numbers.map{String($0)})
    var answer: [Int] = []    
    
    for ar in arrs {
        if sosu(Int(ar)!) && !answer.contains(Int(ar)!) {
            answer.append(Int(ar)!)
        }
    }
    return answer.count
}